qa/rpc-tests/sendtoaddress.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+#
+# This test exercises the sendtoaddress api
+# Tests valid transactions with amounts of different types,
+# Invalid addresses and invalid amounts
+
+from test_framework.test_framework import NavCoinTestFramework
+from test_framework.util import *
+
+
+class SendToAddressTest (NavCoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+
+    def setup_network(self, split=False):
+        self.nodes = start_nodes(self.num_nodes, self.options.tmpdir)
+        connect_nodes_bi(self.nodes,0,1)
+        self.is_network_split=False
+        self.sync_all()
+
+    def add_options(self, parser):
+        parser.add_option("--sendtoaddresstest", dest="sendtoaddresstest",
+                          help="Tests the sendtoaddress api under a variety of inputs")
+
+    def run_test (self):
+        # Turn staking off on all nodes
+        self.nodes[0].staking(False)
+        self.nodes[1].staking(False)
+
+        # Generate necessary NAV for transactions
+        print("Mining blocks...")
+        slow_gen(self.nodes[0], 1)
+        time.sleep(2)
+        self.sync_all()
+        slow_gen(self.nodes[1], 75)
+        self.sync_all()
+        
+        # Assert correct amount of NAV generated
+        assert_equal(self.nodes[0].getbalance(), 59800000)
+        assert_equal(self.nodes[1].getbalance(), 1250)
+        
+        # Make transactions to valid addresses
+        txid0 = self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 60)
+        txid1 = self.nodes[1].sendtoaddress(self.nodes[0].getnewaddress(), Decimal("10.0"))
+        txid7 = self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), "50.0")
+        txid9 = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), 100)
+        self.sync_all()
+        # Generate a block to confirm transactions
+        slow_gen(self.nodes[0], 1)
+        time.sleep(2)
+        self.sync_all()
+        # Assert transactions went through by validating balances
+        assert_equal(self.nodes[0].getbalance(), 59799900 + self.nodes[0].gettransaction(txid0)["fee"] + self.nodes[0].gettransaction(txid7)["fee"] + self.nodes[0].gettransaction(txid9)["fee"])
+        assert_equal(self.nodes[1].getbalance(), 1400 + self.nodes[1].gettransaction(txid1)["fee"])
+
+        # Make transactions with invalid addresses
```

Comment:
We don't check if we actually catch any errors, see the testing pattern I commented elsewhere on tis pr